## Analyzing Shotgun Metagenomic Data

The purpose of this tutorial is to provide one of the possible metagenomic workflows for analyzing shotgun metagenomic next-generation sequencing reads. Keep in mind that there are many different programs that will accomplish each step.

Metagenomics is the study of all genetic material within an environmental sample. Assembly-based metagenomics attempts to assemble the reads from the sample(s) to create contigs and 'bin' each contig into genomes.  This will later allow taxonomical and functional profiling.

This tutorial will perform a comparative genomics study on the human gut microbiome of two populations: Hadza hunter-gatherers in Africa and Western urban Italian adults. This study was performed by Rampelli et al. and all data used in this tutorial was obtained from this study.
Initial Study: [Schnorr et al., 2014](https://christineyanta.github.io/Metagenomics_Tutorial/Schorr2014_HadzaGutMicrobiome.pdf)
Further Study: [Rampelli et al., 2015](https://christineyanta.github.io/Metagenomics_Tutorial/Rampelli2015_HadzaGutMicrobiota.pdf)

To summarize, these two populations differ in their diets. The Hadza diet consists of wild foods, such as meat, honey and berries, that have not undergone cultivation or domestication.  Alternatively, the Italian diet consists entirely of commercial agricultural products that primarily includes plants foods and pasta, along with some dairy and poultry.  


Link to test: [Link text](https://christineyanta.github.io/Metagenomics_Tutorial/Setup/)

